### ISSUE TITLE
Warn or fail when @HollowHashKey is used on unsupported types

### ISSUE THREAD

**Issue by ghost** [Netflix/hollow#186]
The `@HollowHashKey` annotation is supported on `Map` and `Set` types only. However, it's possible to annotate any type and the client API generator will silently ignore it. As it seems reasonable from examples in the docs to, say, apply it to a `List` this can result in confusion or churn later than need be. At least generate a warning when detecting the annotation on a type other than `Map` or `Set`, perhaps with it being a hard failure as an optional config (or fail by default with a mechanism to opt-out).

**Comment by rpalcolea** [comment]
@toolbear,

isn't this for `@HollowHashKey` instead? 

Also, how do you envision this? Something at the `HollowObjectTypeMapper` level to fail if present in a non valid type or a Java `AbstractProcessor` that works on compile time so people could see warnings at compile level on their IDEs? That would take more to get done.

**Comment by akhaku** [comment]
Yup, `@HollowHashKey` is indeed what he meant - updated. What we're going for here is the client code generation - if a data model object has eg a `List` member that is annotated with `@HollowHashKey`, the annotation is ignored silently, and a `findElement` isn't generated.
